openhands/sdk/agent/base.py

```diff
@@ -67,6 +69,38 @@ def agent_context(self) -> AgentContext | None:
         """Returns the agent context used by the Agent."""
         return self._agent_context
 
+    def get_pending_actions(self, state=None) -> list:
+        """Get list of pending actions awaiting confirmation.
+
+        Args:
+            state: ConversationState to scan for unmatched actions.
+                   If None, returns empty list (for backward compatibility).
+        """
+        if state is None:
+            return []
+        return self._get_unmatched_actions(state.events)
+
+    def _get_unmatched_actions(self, events) -> list:
+        """Find actions in the event history that don't have matching observations.
+
+        Optimized to search in reverse chronological order since recent actions
+        are more likely to be unmatched (pending confirmation).
+        """
+
+        observed_action_ids = set()
+        unmatched_actions = []
+
+        # Search in reverse - recent events are more likely to be unmatched
+        for event in reversed(events):
+            if isinstance(event, (ObservationEvent, UserRejectsObservation)):
+                observed_action_ids.add(event.action_id)
+            elif isinstance(event, ActionEvent):
+                if event.id not in observed_action_ids:
+                    # Insert at beginning to maintain chronological order in result
+                    unmatched_actions.insert(0, event)
+
+        return unmatched_actions
```

Comment:
Can we put this as a helper function in `openhands/sdk/event/utils.py` and import it here since it feels it belongs there

Reply:
@openhands please address this feedback